diff.js

```diff
@@ -466,21 +448,34 @@
           // Identical context lines. Track line changes
           if (oldRangeStart) {
             // Close out any changes that have been output (or join overlapping)
-            if (lines.length <= 8 && i < diff.length - 2) {
+            if (lines.length <= options.context * 2 && i < diff.length - 2) {
               // Overlapping
               curRange.push.apply(curRange, contextLines(lines));
             } else {
               // end the range and output
-              var contextSize = Math.min(lines.length, 4);
-              ret.push(
-                  '@@ -' + oldRangeStart + ',' + (oldLine - oldRangeStart + contextSize)
-                  + ' +' + newRangeStart + ',' + (newLine - newRangeStart + contextSize)
-                  + ' @@');
-              ret.push.apply(ret, curRange);
-              ret.push.apply(ret, contextLines(lines.slice(0, contextSize)));
-              if (lines.length <= 4) {
-                eofNL(ret, i, current);
+              var contextSize = Math.min(lines.length, options.context);
+              var hunklines = [];
+              hunklines.push.apply(hunklines, curRange);
```

Comment:
Just do a `curRange.slice()` or even utilize `curRange` if not used elsewhere here.